test/test-barrier.c

```diff
@@ -104,3 +104,38 @@ TEST_IMPL(barrier_3) {
 
   return 0;
 }
+
+static void serial_worker(void* data) {
+  uv_barrier_t* barrier = (uv_barrier_t*) data;
+  if (uv_barrier_wait(barrier) > 0) {
+    uv_barrier_destroy(barrier);
+  }
+  uv_sleep(100);  /* wait a bit before terminating. */
+}
+
+#define NUM_WORKERS 4
+
+/* Ensure that uv_barrier_wait returns positive only after all threads have
+   exited the barrier. If this value is returned too early and the barrier is
+   destroyed prematurely, then this test may see a crash. */
+TEST_IMPL(barrier_serial_thread) {
+  uv_thread_t threads[NUM_WORKERS];
```

Comment:
It'd be slightly more idiomatic to declare this as `uv_thread_t threads[4];` and then use `ARRAY_SIZE(threads)` everywhere.

Reply:
Done.